kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/ModemManagerDbusWrapper.java

```diff
@@ -272,4 +284,84 @@ protected void resetHandlersDisable(String deviceId) {
             }
         }
     }
+
+    protected void failedResetHandlerEnable(String deviceId, Optional<String> modemManagerDbusPath, int delayMinutes)
+            throws DBusException {
+        if (!modemManagerDbusPath.isPresent()) {
+            logger.warn("Cannot retrieve modem device for {}. Skipping modem reset monitor setup.", deviceId);
+            return;
+        }
+
+        Modem mmModemDevice = this.dbusConnection.getRemoteObject(MM_BUS_NAME, modemManagerDbusPath.get(), Modem.class);
+
+        MMFailedModemResetHandler resetHandler = new MMFailedModemResetHandler(mmModemDevice,
+                modemManagerDbusPath.get(), delayMinutes);
+        resetHandler.schedule();
+
+        this.failedModemHandlers.put(deviceId, resetHandler);
+    }
+
+    protected void failedResetHandlersDisable() {
+        for (String deviceId : this.failedModemHandlers.keySet()) {
+            failedResetHandlersDisable(deviceId);
+        }
+        this.modemHandlers.clear();
+    }
+
+    protected void failedResetHandlersDisable(String deviceId) {
+        if (this.failedModemHandlers.containsKey(deviceId)) {
+            MMFailedModemResetHandler handler = this.failedModemHandlers.get(deviceId);
+            handler.cancel();
+        }
+    }
+
+    protected boolean isMMFailedModemResetTimerArmed(String deviceId) {
+        return this.failedModemHandlers.containsKey(deviceId);
+    }
+
+    private class MMFailedModemResetTimerTask extends NMModemResetTimerTask {
+
+        private final String modemManagerDbusPath;
+
+        public MMFailedModemResetTimerTask(Modem modem, String modemManagerDbusPath) {
+            super(modem);
+            this.modemManagerDbusPath = modemManagerDbusPath;
+        }
+
+        @Override
+        public void run() {
+            try {
+                MMModemState modemState = getMMModemState(modemManagerDbusPath);
```

Comment:
Why not use the original variable?

```suggestion
                MMModemState modemState = getMMModemState(this.modem.getObjectPath());
```

Reply:
I cannot use `this.modem` sine this is a private field. However, `this.getModemDbusPath` is available.